pkg/nginxprocess/process.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+// Package nginxprocess contains utilities for working with OS-level NGINX processes.
+package nginxprocess
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/shirou/gopsutil/v4/process"
+)
+
+// Process contains a snapshot of read-only data about an OS-level NGINX process. Create using [List] or [Find].
+type Process struct {
+	Created time.Time // uses millisecond resolution
+	Name    string
+	Cmd     string
+	Exe     string // path to the executable
+	Status  string // process status, only present if this process was created using [WithStatus]
+	PID     int32
+	PPID    int32 // parent PID
+}
+
+// IsWorker returns true if the process is a NGINX worker process.
+func (p *Process) IsWorker() bool { return strings.Contains(p.Cmd, "worker") }
```

Comment:
this isn't good enough, I can get false positives on the master process using `nginx -c /tmp/workers.conf`

Reply:
got this fixed